include/arch/arm/arch/64/mode/object/structures.h

```diff
@@ -54,7 +58,14 @@ typedef word_t vm_rights_t;
 
 #define VCPU_SIZE_BITS      seL4_VCPUBits
 
+#ifdef CONFIG_ARM_PA_SIZE_BITS_40
+/* For a 40 bit PA, EL1 and EL0 use a 3 level translation and skips the PGD */
```

Comment:
Quick question here. 

So generally, this is only true when the kernel is running at EL2, since VTCR_EL2 will be configured to start translations at Level 1. In most other cases, seL4 will run at EL1. These changes make any processor with the 40-bit PA range use a 3-level translation, regardless of kernel execution level. The ZUS+ could run seL4 before, so is there a reason for this setup?

Reply:
You're correct. I haven't experimented to make sure this builds in EL1. The intention of the configs is that `CONFIG_ARM_PA_SIZE_BITS_*` is unused when `CONFIG_ARM_HYPERVISOR_SUPPORT` is not enabled.

Reply:
Alright, so if thats the case, then my commit `fca033c` would need to be updated to only set that cmake variable when hypervisor support is enabled.